File: Sources/CountriesFlagsObjC/source/Countries/UIImage+CoteDIvoire.m
//
//  UIImage+CoteDIvoire.m
//  CountriesFlags
//

#import "Countries/UIImage+CoteDIvoire.h"

@implementation UIImage (CoteDIvoire)

+ (UIImage * _Nullable)flagCoteDIvoireWithSize:(CGSize)size {
    UIColor* colorOrange = [UIColor colorWithRed:1.0f
                                           green:130.0f/255.0f
                                            blue:0.0f
                                           alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    UIColor* colorGreen = [UIColor colorWithRed:0.0f
                                          green:154.0f/255.0f
                                           blue:68.0f/255.0f
                                          alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorOrange setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorWhite setFill];
    UIRectFill(CGRectMake(size.width / 3.0f, 0.0f, size.width / 3.0f, size.height));
    
    [colorGreen setFill];
    UIRectFill(CGRectMake(size.width / 3.0f * 2.0f, 0.0f, size.width / 3.0f, size.height));
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+CoteDIvoire.swift
//
//  UIImage+CoteDIvoire.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Ivory_Coast "wikipedia"
     
     Generates the flag of Côte d'Ivoire.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Côte d'Ivoire.
     */
    internal static func flagCoteDIvoire(with size: CGSize) -> UIImage? {
        let colorOrange = UIColor(red: 255.0/255.0,
                                  green: 130.0/255.0,
                                  blue: 0.0,
                                  alpha: 1.0)
        let colorWhite = UIColor.white
        let colorGreen = UIColor(red: 0.0,
                                 green: 154.0/255.0,
                                 blue: 68.0/255.0,
                                 alpha: 1.0)

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorOrange.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorWhite.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: size.width / 3.0, y: 0.0),
                          size: CGSize(width: size.width / 3.0, height: size.height)))

        colorGreen.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: size.width / 3.0 * 2.0, y: 0.0),
                          size: CGSize(width: size.width / 3.0, height: size.height)))

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
